events/admin.py

```diff
@@ -0,0 +1,82 @@
+from django.contrib import admin
+from django.conf import settings
+
+from .models import Event, Tag, Location
+
+
+@admin.register(Tag)
+class TagAdmin(admin.ModelAdmin):
+    """
+    Admin config for EventTag model
+    """
+
+    list_display = ["name"]
+
+    fields = ["name", "slug"]
+    prepopulated_fields = {"slug": ["name"]}
+
+
+@admin.register(Event)
+class EventAdmin(admin.ModelAdmin):
+    """
+    Admin config for Event model
+    """
+
+    list_display = getattr(
```

Comment:
`ModelAdmin` has a series of methods that get these attrs ([`list_display`](https://github.com/django/django/blob/master/django/contrib/admin/options.py#L947), [`fieldsets`](https://github.com/django/django/blob/master/django/contrib/admin/options.py#L331)) etc.

So a cleaner implementation would be to define the `list_display` etc as normal here. But then override the `get_` methods to return from settings if they exist. For example:
```
list_display = ["title", "start_at", "end_at", "is_published"]

# define other admin attributes as normal......

def get_list_display(self, request):
    if hasattr(settings, "EVENT_ADMIN_LIST_DISPLAY"):
        return settings.EVENT_ADMIN_LIST_DISPLAY

    return super().get_list_display(request)
```
etc

Reply:
As well as making the code cleaner, makes it much easier to test

Reply:
Would have been nice to have known this a while back :wink: I'll take a look